### PR TITLE
Enable jrl-cmakemodules workspace compatiblity

### PR DESCRIPTION
**Pull request by jorisv**

### CODE REVIEW

File: .gitmodules
Comment (cmastalli) [inline .gitmodules:3]: Any reason to do this?

File: CMakeLists.txt
Comment (cmastalli) [inline CMakeLists.txt:68]: Pinocchio is a project dependency. Any reason to do this?

File: unittest/CMakeLists.txt
Comment (cmastalli) [inline unittest/CMakeLists.txt:7]: Same comment here.

File: .gitmodules
Comment (jorisv) [reply .gitmodules:3]: It's linked to this pull request : https://github.com/jrl-umi3218/jrl-cmakemodules/pull/701
submodule will be changed when jrl-cmakemodules PR will be merged.

File: unittest/CMakeLists.txt
Comment (jorisv) [reply unittest/CMakeLists.txt:7]: The only test is a Python test. Then, checking if find_package(pinocchio) have succeed to detect if Pinocchio Python binding will not help.

More generally, I don't think that CMake should check for Python dependencies with CMake modules. Python dependencies should be managed by a package manager (pip, conda, ...).

File: CMakeLists.txt
Comment (cmastalli) [reply CMakeLists.txt:68]: I understand. Imposing this dependency ensures that users have installed Pinocchio from the user side. This is the only requirement where (I think) we should consider here.

**Comment by jorisv** [comment]
@cmastalli Did the modifications are fine to you ?

**Comment by cmastalli** [comment]
> @cmastalli Did the modifications are fine to you ?

I think we should not remove the in https://github.com/Gepetto/example-robot-data/pull/225#discussion_r1679691018. However, I would like to hear from @nim65s on this topic.

**Comment by jorisv** [comment]
@cmastalli I revert the change about add_project_dependency({eigenpy,pinocchio}).

If it's fine to you, you can merge.

File: unittest/CMakeLists.txt
Comment (cmastalli) [reply unittest/CMakeLists.txt:7]: Hi @jorisv!

This is the last comments. Please ping us when it is done! Thanks!

**Comment by jcarpent** [comment]
@cmastalli ping to merge this PR if you are fine.

**Comment by cmastalli** [comment]
> @cmastalli ping to merge this PR if you are fine.

Hi @jcarpent -- We're just missing to address this comment: https://github.com/Gepetto/example-robot-data/pull/225#discussion_r1692859261.

This should take no time to do it.

File: unittest/CMakeLists.txt
Comment (cmastalli) [reply unittest/CMakeLists.txt:7]: Could you provide details on why those cases fail? In any case, I'll let @nim65s decide this. He is more competent in judging your suggestion. Thanks!

Comment (nim65s) [reply unittest/CMakeLists.txt:7]: We only have python unittest, and when python is active, there is a `add_project_dependency(pinocchio 2.7.0 REQUIRED)`, so the only case where things could fail would be if someone installs pinocchio without python interface and try to run example-robot-data python unittest.
This case can be dismissed, so I think we can remove this error message.

**Review by nim65s — APPROVED** [review]

**Pull request merged**
